server/core/src/main/java/org/infinispan/server/core/transport/NettyChannelInitializer.java

```diff
@@ -0,0 +1,90 @@
+   protected final ProtocolServer<A> server;
+   protected final NettyTransport transport;
+   protected final ChannelOutboundHandler encoder;
+
+   public NettyChannelInitializer(ProtocolServer<A> server, NettyTransport transport, ChannelOutboundHandler encoder) {
+      this.server = server;
+      this.transport = transport;
+      this.encoder = encoder;
+   }
+
+   @Override
+   protected void initChannel(Channel ch) throws Exception {
+      ChannelPipeline pipeline = ch.pipeline();
+      pipeline.addLast("stats", new StatsChannelHandler(transport));
+      SslConfiguration ssl = server.getConfiguration().ssl();
+      if (ssl.enabled()) {
+         //add default domain mapping
+         JdkSslContext defaultNettySslContext = createNettySslContext(ssl, ssl.sniDomainsConfiguration().get("*"));
+         DomainMappingBuilder<JdkSslContext> domainMappingBuilder = new DomainMappingBuilder<>(defaultNettySslContext);
+
+         //and the rest
+         ssl.sniDomainsConfiguration().forEach((k, v) -> {
+            if (!"*".equals(k)) {
+               domainMappingBuilder.add(k, createNettySslContext(ssl, v));
+            }
+         });
+
+         pipeline.addLast("sni", new SniHandler(domainMappingBuilder.build()));
+      }
+      pipeline.addLast("decoder", server.getDecoder());
+      if (encoder != null)
+         pipeline.addLast("encoder", encoder);
+   }
+
+   private JdkSslContext createNettySslContext(SslConfiguration sslConfiguration, SslEngineConfiguration sslEngineConfiguration) {
+      SSLContext sslContext;
+      if (sslEngineConfiguration.sslContext() != null) {
+         sslContext = sslEngineConfiguration.sslContext();
+      } else {
+         sslContext = SslContextFactory.getContext(sslEngineConfiguration.keyStoreFileName(),
+                 sslEngineConfiguration.keyStorePassword(), sslEngineConfiguration.keyStoreCertificatePassword(),
+                 sslEngineConfiguration.trustStoreFileName(), sslEngineConfiguration.trustStorePassword());
+      }
+      return createSslContext(sslContext, requireClientAuth(sslConfiguration));
+   }
+
+  private JdkSslContext createSslContext(SSLContext sslContext, ClientAuth clientAuth) {
+    //Unfortunately we need to grap a list of available ciphers from the engine.
+    //If we won't, JdkSslContext will use common ciphers from DEFAULT and SUPPORTED, which gives us 5 out of ~50 available ciphers
+    //Of course, we don't need to any specific engine configuration here... just a list of ciphers
+    String[] ciphers = SslContextFactory.getEngine(sslContext, false, false).getSupportedCipherSuites();
+    return new JdkSslContext(sslContext, false, Arrays.asList(ciphers), IdentityCipherSuiteFilter.INSTANCE, null, clientAuth);
+  }
+
+   private ClientAuth requireClientAuth(SslConfiguration sslConfig) {
+      return sslConfig.enabled() ? ClientAuth.REQUIRE : ClientAuth.NONE;
```

Comment:
Aha, thanks :) - I figured it was in NettyChannelInitializer somewhere, not sure how I didn't see this one.  Thanks!